bpf/kmesh/workload/include/authz.h

```diff
@@ -182,13 +205,278 @@ static int match_dst_ports(Istio__Security__Match *match, struct xdp_info *info,
+        if (!notIpPtrs) {
+            return UNMATCHED;
+        }
+
+#pragma unroll
+        for (i = 0; i < MAX_MEMBER_NUM_PER_POLICY; i++) {
+            if (i >= params->n_not_ips) {
+                break;
+            }
+
+            if (bpf_probe_read_kernel(&notIpAddr, sizeof(notIpAddr), &notIpPtrs[i]) != 0) {
+                continue;
+            }
+
+            if (!notIpAddr) {
+                continue;
+            }
+
+            notIp = (Istio__Security__Address *)KMESH_GET_PTR_VAL((void *)notIpAddr, Istio__Security__Address);
+            if (!notIp) {
+                continue;
+            }
+
+            if (match_ip_rule(&notIp->address, notIp->length, params->tuple_info, params->ip_type) == MATCHED) {
+                return UNMATCHED;
+            }
+        }
+    }
+
+    // Match IPs
+    if (params->n_ips != 0) {
+        ipPtrs = KMESH_GET_PTR_VAL(params->ip_list, void *);
+        if (!ipPtrs) {
+            return UNMATCHED;
+        }
+
+#pragma unroll
+        for (i = 0; i < MAX_MEMBER_NUM_PER_POLICY; i++) {
+            if (i >= params->n_ips) {
+                break;
+            }
+
+            if (bpf_probe_read_kernel(&ipAddr, sizeof(ipAddr), &ipPtrs[i]) != 0) {
+                continue;
+            }
+
+            if (!ipAddr) {
+                continue;
+            }
+
+            ip = (Istio__Security__Address *)KMESH_GET_PTR_VAL((void *)ipAddr, Istio__Security__Address);
+            if (!ip) {
+                continue;
+            }
+
+            if (match_ip_rule(&ip->address, ip->length, params->tuple_info, params->ip_type) == MATCHED) {
+                return MATCHED;
+            }
+        }
+    }
+    return UNMATCHED;
+}
+
+static inline int match_src_ip(Istio__Security__Match *match, struct bpf_sock_tuple *tuple_info)
+{
+    if (!match || !tuple_info) {
+        return UNMATCHED;
+    }
+
+    struct MatchIpParams params = {
+        .match = match,
+        .tuple_info = tuple_info,
+        .ip_list = match->source_ips,
+        .not_ip_list = match->not_source_ips,
+        .n_ips = match->n_source_ips,
+        .n_not_ips = match->n_not_source_ips,
+        .ip_type = TYPE_SRCIP,
+    };
+    return match_ip_common(&params);
+}
+
+static inline int match_dst_ip(Istio__Security__Match *match, struct bpf_sock_tuple *tuple_info)
+{
+    if (!match || !tuple_info) {
+        return UNMATCHED;
+    }
+
+    struct MatchIpParams params = {
+        .match = match,
+        .tuple_info = tuple_info,
+        .ip_list = match->destination_ips,
+        .not_ip_list = match->not_destination_ips,
+        .n_ips = match->n_destination_ips,
+        .n_not_ips = match->n_not_destination_ips,
+        .ip_type = TYPE_DSTIP,
+    };
+    return match_ip_common(&params);
+}
+
+static inline int match_IPs(Istio__Security__Match *match, struct bpf_sock_tuple *tuple_info)
+{
+    return match_src_ip(match, tuple_info) || match_dst_ip(match, tuple_info);
```

Comment:
How can we match both src and dst ip address. if src one matched, donot we need to match dst too?

Reply:
Here should use &&

Reply:
done